src/useChat.test.ts

```diff
@@ -134,9 +144,176 @@ describe("useChat", () => {
     });
   });
 
+  describe("agentic behavior", () => {
+    (globalThis as any).fetch ||= jest.fn();
+    const global = globalThis as typeof globalThis & { fetch: any };
```

Comment:
I think the following might be a more simpler equivalent of this, while also avoiding the use of `any`. Correct me if I'm wrong though.
```suggestion
    const global = { ...globalThis, fetch: globalThis.fetch ?? jest.fn() }
```

Reply:
So unfortunately, we'll need to keep the original way of writing it.
In the context of jest, `globalThis` is essentially the global namespace (sort of like `window` in the browser context). As such, on line 148 we're mocking `fetch` in that global namespace.

With the change you've suggested, we're defining a new object called global and that's where we are providing the mock of `fetch`. So the problem there is that the global namespace doesn't get the fetch mock- only this new object which we've defined called `global`. 

Furthermore, the only reason I originally defined `global` on line 149 was to be able to avoid casting it to that type below- it looked really messy to have to type:

```
const mockFetch = jest.spyOn(globalThis as typeof globalThis & { fetch: any }, "fetch").mockResolvedValue({
  status: 200,
  json: jest.fn().mockResolvedValue({ event: "mock-event" })
});

```